include/LTL/Structures/BitProductStateSet.h

```diff
@@ -97,13 +85,18 @@ namespace LTL { namespace Structures {
             ++_discovered;
             const auto res = _markings.add(state);
             if (res.second == std::numeric_limits<size_t>::max()) {
-                return res;
+                return {res.first, res.second, res.second};
```

Comment:
Would be nice to have some comment explaining the third field here (perhaps in the docstring for this function), even after some digging I am not sure I understand the purpose/meaning of it.